File: ClassM8_Client/ClassM8_Client/Data/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;
using System.Threading.Tasks;

namespace ClassM8_Client.Data
{
    [DataContract]
    class Message
    {

        [DataMember]
        private String content;

        [DataMember]
        private String sender;

        [DataMember]
        public String dateTime;

        public Message() { }

        public Message(String c, String s, String dt)
        {
            this.content = c;
            this.sender = s;
            this.dateTime = dt;
        }

        public String getContent()
        {
            return content;
        }

        public void setContent(String content)
        {
            this.content = content;
        }

        public String getSender()
        {
            return sender;
        }

        public void setSender(String sender)
        {
            this.sender = sender;
        }

        public String getDateTime()
        {
            return dateTime;
        }

        public void setDateTime(String dateTime)
        {
            this.dateTime = dateTime;
        }

        public bool Equals(Message m)
        {
            if (m == null)
            {
                return false;
            }
            return (this.content.Equals(m.content)) && (this.sender.Equals(m.sender)) && (this.dateTime == m.dateTime);
        }

        public override string ToString()
        {
            String time = this.getDateTime().Split('T')[1];
            time = time.Split('.')[0];
            return this.getSender() + "   " + time + "\r\n" + this.getContent(); ;
        }

        internal string getFormattedDate()
        {
            String ret = "";
            ret = this.dateTime.Replace("T", " ");
            ret = ret.Substring(0, ret.Length - 13);
            return ret;
        }
    }
}

File: Android/backend/src/main/java/com/example/backend/Dto/Message.java
package com.example.backend.Dto;

import java.sql.Timestamp;
import java.util.Date;

public class Message {

    private String content;

    private String sender;

    private Timestamp dateTime;

    public Message() {}

    public Message(String content, String sender, Timestamp dateTime) {
        this.content = content;
        this.sender = sender;
        this.dateTime = dateTime;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getSender() {
        return sender;
    }

    public void setSender(String sender) {
        this.sender = sender;
    }

    public Timestamp getDateTime() {
        return dateTime;
    }

    public void setDateTime(Timestamp dateTime) {
        this.dateTime = dateTime;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Message message = (Message) o;

        if (content != null ? !content.equals(message.content) : message.content != null)
            return false;
        if (sender != null ? !sender.equals(message.sender) : message.sender != null) return false;
        return dateTime != null ? dateTime.equals(message.dateTime) : message.dateTime == null;

    }

    @Override
    public int hashCode() {
        int result = content != null ? content.hashCode() : 0;
        result = 31 * result + (sender != null ? sender.hashCode() : 0);
        result = 31 * result + (dateTime != null ? dateTime.hashCode() : 0);
        return result;
    }

    @Override
    public String toString() {
        return "Message{" +
                "content='" + content + '\'' +
                ", sender='" + sender + '\'' +
                ", dateTime=" + dateTime +
                '}';
    }
}
